_static/theme/zammad_overrides.js

```diff
@@ -0,0 +1,76 @@
+/* Custom overrides */
+/* No oversized <pre>, please - Thanks to @TJemxx */
+function shortPre(i) {
+   var attr = 'pre-' + i
+   var pre = document.getElementById(attr);
+   var button = document.getElementById('button-' + attr);
+
+   if ( pre.classList.contains('shorten-pre')) {
+      pre.classList.remove('shorten-pre');
+      pre.classList.add('release-shorten-pre');
+
+      $(button).html('Reduce to smaller size ...')
+   } else {
+      pre.classList.remove('release-shorten-pre');
+      pre.classList.add('shorten-pre');
+
+      $(button).html('Expand to full size ...')
+   }
+}
+
+// Update <pre> elements
+function WorkPres(item, index) {
+   var size = 250;
+
+   if ( !document.getElementById(item) ){
+      item.setAttribute('id', 'pre-' + index);
+   }
+
+   if( item.clientHeight > size ){
+      
+      if ( !document.getElementById('button-' + item) ){
+         var button = document.createElement('button');
+         button.setAttribute('onclick', 'shortPre(' + index + ')');
+         button.setAttribute('id', 'button-pre-' + index);
+         $(button).html('Expand to full size ...').addClass('shorten-pre-button');      
+         item.appendChild(button);
+
+         console.log("added button for big <pre> (pre-" + index + ")")
+      }
+      
+      item.classList.add('shorten-pre');
+   }
+};
+
+// Provide all tabs with onClick action
+function WorkTabs(item) {
+   item.setAttribute('onclick', 'CalculatePres()');
+}
+
+// Function to fire (re)-calculation for <pre> elements
+function CalculatePres(){
+   pres = document.getElementsByTagName('pre');
+   for(var i = 0; i < pres.length; i++){ 
+      WorkPres(pres[i], i);
+   };
+   console.log("<pre> elements reviewed ...")
+}
+
+function AddOnClick(){
+   // Find all possible tabs to add onClick actions
+   console.log("Add onClick actions for <a>-data-tabs (if any) ...")
+   tabs = document.getElementsByTagName('a');
+   for(var i = 0; i < tabs.length; i++){ 
+      if ( tabs[i].hasAttribute('data-tab') ){
+         WorkTabs(tabs[i]);
+      }
+   };
+}

```

Comment:
Yes you are missing required information to understand as of why this appeared in the code!

Context:
I'm currently working on a quite big refactor of the installation parts of the system admin documentation.
In these steps I discovered `sphinx-tabs` to be very useful for reducing several pages into one by keeping contexts in tact.
Most of our package installations are absolutely identical apart from commands that are OS specific.

Now, in order to be able to measure the height of an element (which is what function `WorkPres()` is responsible for) the element has to be visible. In early steps and attempts I did try to simply temporary set non-active tabs to `display: block` and remove the attribute after measuring.

The issue was, this was prooving to be inefficient and to some parts failing randomly.
I thus decided that the re-measuring should happen if required - this is why above function exists.

I did use the js-code of the admin-docs for the docs (because it was there first) and extended it. 
After extending it I brought back the changes to admin-docs as well.

This allows us to use `sphinx-tabs` in the future without changing anything else.
This is why it contains code you don't need in this specific case.